lib/src/widgets/radio/ds_radio.widget.dart

```diff
@@ -0,0 +1,321 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:blip_ds/blip_ds.dart';
+import 'package:flutter/material.dart';
+
+/// A material design radio button custom.
+///
+/// Used to select between a number of mutually exclusive values. When one radio
+/// button in a group is selected, the other radio buttons in the group cease to
+/// be selected. The values are of type `T`, the type parameter of the [Radio]
+/// class. Enums are commonly used for this purpose.
+///
+/// The radio button itself does not maintain any state. Instead, selecting the
+/// radio invokes the [onChanged] callback, passing [value] as a parameter. If
+/// [groupValue] and [value] match, this radio will be selected. Most widgets
+/// will respond to [onChanged] by calling [State.setState] to update the
+/// radio button's [groupValue].
+///
+/// {@tool dartpad}
+/// Here is an example of Radio widgets wrapped in ListTiles, which is similar
+/// to what you could get with the RadioListTile widget.
+///
+/// The currently selected character is passed into `groupValue`, which is
+/// maintained by the example's `State`. In this case, the first `Radio`
+/// will start off selected because `_character` is initialized to
+/// `SingingCharacter.lafayette`.
+///
+/// If the second radio button is pressed, the example's state is updated
+/// with `setState`, updating `_character` to `SingingCharacter.jefferson`.
+/// This causes the buttons to rebuild with the updated `groupValue`, and
+/// therefore the selection of the second button.
+///
+/// Requires one of its ancestors to be a [Material] widget.
+///
+/// ** See code in examples/api/lib/material/radio/radio.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [RadioListTile], which combines this widget with a [ListTile] so that
+///    you can give the radio button a label.
+///  * [Slider], for selecting a value in a range.
+///  * [Checkbox] and [Switch], for toggling a particular value on or off.
+///  * <https://material.io/design/components/selection-controls.html#radio-buttons>
+class DSRadio<T> extends Radio<T> {
+  /// Creates a material design radio button.
+  ///
+  /// The radio button itself does not maintain any state. Instead, when the
+  /// radio button is selected, the widget calls the [onChanged] callback. Most
+  /// widgets that use a radio button will listen for the [onChanged] callback
+  /// and rebuild the radio button with a new [groupValue] to update the visual
+  /// appearance of the radio button.
+  ///
+  /// The following arguments are required:
+  ///
+  /// * [value] and [groupValue] together determine whether the radio button is
+  ///   selected.
+  /// * [onChanged] is called when the user selects this radio button.
+  const DSRadio({
+    Key? key,
+    required super.value,
+    required super.groupValue,
+    required super.onChanged,
+    super.mouseCursor,
```

Comment:
I think keeping all possible resources on the original radio button opens up more possibilities for future reuse. It would be unproductive to limit this.

The DSRadioTile widget, which will be commonly used, only contemplates these parameters.

Reply:
Done

Reply:
Done

Reply:
adjusted parameters

Reply:
Done